src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+            command = scanner.nextLine();
+            parser.interpretCommand(command);
+            String firstCommand = this.parser.getFirstCommand();
+            try {
+                switch (firstCommand) {
+                case "bye":
+                    this.ui.goodBye();
+                    stillRunning = false;
+                    break;
+                case "done":
+                    markDone();
+                    break;
+                case "deadline":
+                case "todo":
+                case "event":
+                    addTask();
+                    break;
+                case "delete":
+                    deleteTask();
+                    break;
+                case "list":
+                    this.tasks.listTasks();
+                    break;
+                case "find":
+                    String keyword = this.parser.findKeyword();
+                    ArrayList<Task> tasksWithKeyword = this.tasks.findTasksUsingKeyword(keyword);
+                    this.ui.showTasksWithKeyword(tasksWithKeyword);
+                    break;
+                default:
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
+            } catch (DukeException e) {
+                this.ui.showError(e);
+            }
+        }
+    }
+
+    /**
+     * Method to delete task.
+     *
+     * @throws DukeException Thrown whenever user requests delete of a
+     * task out of range or not a number. eg. <code>delete hi</code>
+     */
+    public void deleteTask() throws DukeException {
+        try {
+            this.tasks.deleteTask(parser.findCommandIndex());
+            this.ui.showDeleteTaskMessage(this.tasks.getTasksLength());
+            this.writeDataToDuke();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! Index out of range!");
+        } catch (NumberFormatException e) {
+            throw new DukeException("☹ OOPS!!! Put a number after 'delete'!");
+        }
+    }
+
+    private void findTasks() {
+        String keyword = parser.findKeyword();
+
+    }
+
+    private void writeDataToDuke() {
+        this.storage.writeData(this.tasks);
+    }
+
+    /**
+     * Method to add task to duke.Duke.
+     *
+     * @throws DukeException Thrown when the task is not given a description
+     * or when the user does not give a date for an event or deadline task,
+     * or when the user formats the date wrongly.
+     */
+    public void addTask() throws DukeException {
+        String firstCommand = this.parser.getFirstCommand();
+        String date = this.parser.findDateInCommand();
+        String taskDesc = this.parser.findTaskDescription();
+        String aOrAn = firstCommand.equals("event") ? "an" : "a";
+        if (taskDesc.equals("")) {
+            throw new DukeException("☹ OOPS!!! The description of " + aOrAn + " " + firstCommand + " cannot be empty.");
```

Comment:
Should these error messages be abstracted into final Strings so that they can be easily modified when the errors are thrown? For example, you could consider having methods in the ```DukeException``` class to return a ```DukeException``` instance with a specific type of error message:

```java
public class DukeException extends Exception {
    public DukeException(String message) {
      super(message);
    }
    //...
    public static final String EMPTY_DESC_ERROR = "Empty description error!";
    //...
    public static DukeException emptyDescError() {
      return new DukeException(EMPTY_DESC_ERROR);
    }
}
```
Then, throw the exception with
```throw DukeException.emptyDescError();```